src/osx_clipboard.rs

```diff
@@ -45,14 +45,23 @@ impl OSXClipboardContext {
 
 impl ClipboardProvider for OSXClipboardContext {
     fn get_contents(&mut self) -> Result<String> {
-        let nsstring: *mut NSString =
-            unsafe { msg_send![self.pasteboard, stringForType: NSPasteboardTypeString] };
-        if nsstring.is_null() {
-            Err("pasteboard#stringForType returned null".into())
-        } else {
-            let nsstring: Id<NSString> = unsafe { Id::from_retained_ptr(nsstring) };
```

Comment:
The issue in particular was using `Id::from_retained_ptr`; it should have been `Id::from_ptr`